### ISSUE TITLE
Input is not being validated on the business logic side

### ISSUE THREAD

**Issue by adrianbranescu** [Adrian-s-JavaScript-Programs/angular-locomotive-check-digit-generator#4]
Currently, input validation is only done on user-side, but that can be bypassed.
Through small workarounds, invalid data can be passed to the application, leading to undesired results.

Example:

Open the application. Right click on input field and choose "Inspect Element":
![1](https://user-images.githubusercontent.com/59761765/112736858-a43b9c80-8f5e-11eb-99ae-846b69fe30c5.png)

You will see "maxlength" attribute, which is set to be 11:
![2](https://user-images.githubusercontent.com/59761765/112736860-a4d43300-8f5e-11eb-97b0-d7fc8be89a13.png)

Using browser inspector, you can change this attribute. For example, you can eliminate it:
![3](https://user-images.githubusercontent.com/59761765/112736861-a4d43300-8f5e-11eb-8a5e-6a507e2c66f8.png)

After that, you can enter more than 11 characters in the input field:
![4](https://user-images.githubusercontent.com/59761765/112736862-a56cc980-8f5e-11eb-9a0f-699a2743f51d.png)

Because form is invalid, "Generate" button is disabled by the application. But that can also be changed. Right click on "Generate" button and choose "Inspect Element":
![5](https://user-images.githubusercontent.com/59761765/112736863-a56cc980-8f5e-11eb-82a7-a2c3cb716c88.png)

You will see "Generate" button has "disabled" attribute:
![6](https://user-images.githubusercontent.com/59761765/112736864-a6056000-8f5e-11eb-98f2-c12ec5a3b21a.png)

When present, "disabled" attribute makes the element unusable, regardless of the value the attribute has. You cannot enable the submit button by setting a boolean value. Instead, you must remove the "disabled" attribute.

After the deletion of "disabled" attribute, "Generate" button becomes enabled:
![7](https://user-images.githubusercontent.com/59761765/112736865-a6056000-8f5e-11eb-848a-1522af911c58.png)

You can now press "Generate" button, which will submit an invalid form that will be processed incorrectly:
![8](https://user-images.githubusercontent.com/59761765/112736867-a69df680-8f5e-11eb-9574-fd439c6fb88f.png)

In conclusion, additional input validation on the business-side must be made. Check digit calculation should be done only after making sure input is valid. Otherwise, an error message should be displayed to the user.

**Comment by adrianbranescu** [comment]
Fix tested with Mozilla Firefox:

![Mozilla Firefox](https://user-images.githubusercontent.com/59761765/113209723-12909f80-927c-11eb-9bb4-edbcdd9adb99.png)

Fix tested with Google Chrome:

![Google Chrome](https://user-images.githubusercontent.com/59761765/113209785-25a36f80-927c-11eb-90c9-9e3212cd2ef5.png)
